### PR TITLE
Daravind/issue 1062

### PR DESCRIPTION
**Pull request by deepakaravindr**
Fixes https://github.com/NuGet/Home/issues/1098
and adds test for
https://github.com/NuGet/Home/issues/1062

@yishaigalatzer @emgarten @pranavkm

### CODE REVIEW

File: src/NuGet.Configuration/Utility/SettingsUtility.cs [hunk 1]
```diff
@@ -130,16 +130,16 @@ public static string GetGlobalPackagesFolder(ISettings settings)
                 throw new ArgumentNullException(nameof(settings));
             }
 
-            // Note that the value GlobalPackagesFolder must be a full path and not a relative path
-            // Read it like a string and explicitly not as a path
-            var path = settings.GetValue(ConfigSection, GlobalPackagesFolderKey, isPath: false);
+            var path = Environment.GetEnvironmentVariable(GlobalPackagesFolderEnvironmentKey);
             if (!string.IsNullOrEmpty(path))
             {
                 path = path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
```
Comment (emgarten) [inline src/NuGet.Configuration/Utility/SettingsUtility.cs:136]: The replace/return should be done in one place, these can both use the same code path once it picks where to get the value from.

**Comment by emgarten** [comment]
:shipit:

File: test/NuGet.Configuration.Test/SettingsTests.cs [hunk 2]
```diff
@@ -1791,6 +1791,31 @@ public void GetGlobalPackagesFolder_FromNuGetConfig()
...
+</configuration>";
+
+            var nugetConfigPath = "NuGet.config";
+            using (var mockBaseDirectory = TestFilesystemUtility.CreateRandomTestFolder())
+            {
+                TestFilesystemUtility.CreateConfigurationFile(nugetConfigPath, mockBaseDirectory, config);
+                Settings settings = new Settings(mockBaseDirectory);
+
+                // Act
+                var globalPackagesFolderPath = SettingsUtility.GetGlobalPackagesFolder(settings);
```
Comment (pranavkm) [inline test/NuGet.Configuration.Test/SettingsTests.cs:1811]: Does the calling code always `Path.GetFullPath(Path.Combine(Directory.GetCurrentDirectory(), SettingsUtility.GetGlobalPackagesFolder(settings)))`?

File: src/NuGet.Configuration/Utility/SettingsUtility.cs (hunk 1 above)
Comment (pranavkm) [inline src/NuGet.Configuration/Utility/SettingsUtility.cs:136]: Why do you need to do the replace at all? `Path` operations work well with either separators?

Comment (deepakaravindr) [inline src/NuGet.Configuration/Utility/SettingsUtility.cs:136]: We need to replace it. We are reading in the value as a string, not as a path. Note that we have always done this for repositoryPath

File: test/NuGet.Configuration.Test/SettingsTests.cs (hunk 2 above)
Comment (deepakaravindr) [inline test/NuGet.Configuration.Test/SettingsTests.cs:1811]: No. With https://github.com/NuGet/Home/issues/1062, if globalPackagesFolder is relative, full path is constructed relative to solution directory. exe throws if solution directory is not available (Either the switch or.sln file)

File: test/NuGet.Configuration.Test/SettingsTests.cs [hunk 3]
```diff
@@ -1791,6 +1791,31 @@ public void GetGlobalPackagesFolder_FromNuGetConfig()
...
+
+                // Assert
+                Assert.Equal(@"..\..\NuGetPackages", globalPackagesFolderPath);
```
Comment (yishaigalatzer) [inline test/NuGet.Configuration.Test/SettingsTests.cs:1814]: is this test going to pass on Mono?

**Comment by yishaigalatzer** [comment]
Can this PR be closed?

**Comment by deepakaravindr** [comment]
Yes

**Pull request closed without merging**